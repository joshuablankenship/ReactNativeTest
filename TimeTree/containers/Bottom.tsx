import * as React from 'react';
import {Text, View, TouchableOpacity, Image} from 'react-native';
import {styles} from '../styles/BottomStyles';
import RBSheet from 'react-native-raw-bottom-sheet';
import {useRef} from 'react';
import Signup from './Signup';

type Props = {
  navigation: any;
  closeLoginSheet: () => void;
};
export default function Bottom(props: Props) {
  const signupSheet: RBSheet | null | undefined | any = useRef();

  const _openSignUpSheet = () => {
    signupSheet.current.open();
  };
  const _closeSignUpSheet = () => {
    signupSheet.current.close();
  };
  return (
    <View style={styles.buttonContainer}>
      <TouchableOpacity style={styles.buttonGreen}>
        <Text style={styles.buttonText}>Sign in via email</Text>
      </TouchableOpacity>
      <TouchableOpacity style={styles.buttonFacebook}>
        <Image
          style={styles.f_logo}
          source={require('../assets/images/facebook_logo.png')}
        />
        <Text style={styles.buttonText}> Login with Facebook</Text>
      </TouchableOpacity>
      <TouchableOpacity style={styles.buttonApple}>
        <Image
          style={styles.a_logo}
          source={require('../assets/images/apple_logo.png')}
        />
        <Text style={styles.buttonText}> Sign in with Apple</Text>
      </TouchableOpacity>
      <View style={styles.textContainer}>
        <Text style={styles.text}>
          If you don't have an account,{' '}
          <Text style={styles.link} onPress={_openSignUpSheet}>
            create one
          </Text>
          .
        </Text>
      </View>
      <View style={styles.container}>
        <RBSheet
          ref={signupSheet}
          height={800}
          openDuration={250}
          customStyles={{
            container: {
              borderTopLeftRadius: 10,
              borderTopRightRadius: 10,
            },
          }}>
          <Signup
            openSignupSheet={_openSignUpSheet}
            closeSignupSheet={_closeSignUpSheet}
            closeLoginSheet={props.closeLoginSheet}
            navigation={props.navigation}
          />
        </RBSheet>
      </View>
    </View>
  );
}
