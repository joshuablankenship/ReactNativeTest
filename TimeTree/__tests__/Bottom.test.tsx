// eslint-disable-next-line @typescript-eslint/no-unused-vars
import {shallow, ShallowWrapper} from 'enzyme';
import React from 'react';
import Bottom from '../containers/Bottom';
// const SplashScreen = jest.mock('react-native-splash');
describe('Test Login Screen', () => {
  let wrapper: ShallowWrapper;
  it('It should match initial snapshot', () => {
    wrapper = shallow(<Bottom />);
    expect(wrapper).toMatchSnapshot();
  });
  //   // it('It should display Login with Google button', () => {
  //   //   props = createTestProps;
  //   //   wrapper = shallow(<Login {...props} />);
  //   //   expect(wrapper.find(TouchableOpacity).render().text()).toBe(
  //   //     ' Login with Google',
  //   //   );
  //   // });
  // });
  // it('renders correctly', () => {
  //   renderer.create(<Login />);
});
