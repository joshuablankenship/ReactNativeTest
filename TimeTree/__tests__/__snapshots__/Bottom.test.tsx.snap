// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Bottom Screen It should match initial snapshot 1`] = `
<View
  style={
    Object {
      "marginLeft": 40,
      "marginRight": 40,
      "marginTop": 10,
    }
  }
>
  <ForwardRef
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#2FCC87",
        "borderRadius": 5,
        "color": "white",
        "flexDirection": "row",
        "justifyContent": "space-evenly",
        "marginBottom": 10,
        "padding": 12,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontWeight": "bold",
        }
      }
    >
      Sign in via email
    </Text>
  </ForwardRef>
  <ForwardRef
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#3A5A97",
        "borderRadius": 5,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 10,
        "padding": 12,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/images/facebook_logo.png",
        }
      }
      style={
        Object {
          "height": 25,
          "width": 25,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "white",
          "fontWeight": "bold",
        }
      }
    >
       Login with Facebook
    </Text>
  </ForwardRef>
  <ForwardRef
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "black",
        "borderRadius": 5,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 10,
        "padding": 12,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/images/apple_logo.png",
        }
      }
      style={
        Object {
          "height": 15,
          "width": 15,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "white",
          "fontWeight": "bold",
        }
      }
    >
       Sign in with Apple
    </Text>
  </ForwardRef>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      If you don't have an account,
       
      <Text
        onPress={[Function]}
        style={
          Object {
            "alignItems": "center",
            "color": "black",
            "flex": -1,
            "flexDirection": "row",
            "fontWeight": "bold",
            "justifyContent": "center",
            "marginTop": 5,
          }
        }
      >
        create one
      </Text>
      .
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "paddingTop": 200,
      }
    }
  >
    <RBSheet
      animationType="none"
      closeDuration={200}
      closeOnDragDown={false}
      closeOnPressBack={true}
      closeOnPressMask={true}
      customStyles={
        Object {
          "container": Object {
            "borderTopLeftRadius": 10,
            "borderTopRightRadius": 10,
          },
        }
      }
      dragFromTopOnly={false}
      height={800}
      keyboardAvoidingViewEnabled={true}
      minClosingHeight={0}
      onClose={null}
      onOpen={null}
      openDuration={250}
    >
      <Signup
        closeSignupSheet={[Function]}
        openSignupSheet={[Function]}
      />
    </RBSheet>
  </View>
</View>
`;

exports[`Test Login Screen It should match initial snapshot 1`] = `
<View
  style={
    Object {
      "marginLeft": 40,
      "marginRight": 40,
      "marginTop": 10,
    }
  }
>
  <ForwardRef
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#2FCC87",
        "borderRadius": 5,
        "color": "white",
        "flexDirection": "row",
        "justifyContent": "space-evenly",
        "marginBottom": 10,
        "padding": 12,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontWeight": "bold",
        }
      }
    >
      Sign in via email
    </Text>
  </ForwardRef>
  <ForwardRef
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#3A5A97",
        "borderRadius": 5,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 10,
        "padding": 12,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/images/facebook_logo.png",
        }
      }
      style={
        Object {
          "height": 25,
          "width": 25,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "white",
          "fontWeight": "bold",
        }
      }
    >
       Login with Facebook
    </Text>
  </ForwardRef>
  <ForwardRef
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "black",
        "borderRadius": 5,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 10,
        "padding": 12,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/images/apple_logo.png",
        }
      }
      style={
        Object {
          "height": 15,
          "width": 15,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "white",
          "fontWeight": "bold",
        }
      }
    >
       Sign in with Apple
    </Text>
  </ForwardRef>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      If you don't have an account,
       
      <Text
        onPress={[Function]}
        style={
          Object {
            "alignItems": "center",
            "color": "black",
            "flex": -1,
            "flexDirection": "row",
            "fontWeight": "bold",
            "justifyContent": "center",
            "marginTop": 5,
          }
        }
      >
        create one
      </Text>
      .
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "paddingTop": 200,
      }
    }
  >
    <RBSheet
      animationType="none"
      closeDuration={200}
      closeOnDragDown={false}
      closeOnPressBack={true}
      closeOnPressMask={true}
      customStyles={
        Object {
          "container": Object {
            "borderTopLeftRadius": 10,
            "borderTopRightRadius": 10,
          },
        }
      }
      dragFromTopOnly={false}
      height={800}
      keyboardAvoidingViewEnabled={true}
      minClosingHeight={0}
      onClose={null}
      onOpen={null}
      openDuration={250}
    >
      <Signup
        closeSignupSheet={[Function]}
        openSignupSheet={[Function]}
      />
    </RBSheet>
  </View>
</View>
`;
